PurchasesCoreSwift/Networking/HTTPClient.swift

```diff
@@ -0,0 +1,306 @@
+                        requestBody maybeRequestBody: [String: Any]?,
+                        authHeaders: [String: String],
+                        retried: Bool = false,
+                        completionHandler maybeCompletionHandler: ((Int, [AnyHashable: Any]?, Error?) -> Void)?) {
+        operationDispatcher.dispatchOnHTTPSerialQueue { [self] in
+            do {
+                try self.assertIsValidRequest(httpMethod: httpMethod, requestBody: maybeRequestBody)
+            } catch let error {
+                if let completionHandler = maybeCompletionHandler {
+                    self.operationDispatcher.dispatchOnMainThread {
+                        completionHandler(-1, nil, error)
+                    }
+                    return
+                }
+            }
+
+            let requestHeaders = self.defaultHeaders.merging(authHeaders, uniquingKeysWith: { (_, last) in last })
+
+            let maybeURLRequest = self.createRequest(httpMethod: httpMethod, path: path, requestBody: maybeRequestBody,
+                                                     headers: requestHeaders, refreshETag: retried)
+
+            guard let urlRequest = maybeURLRequest else {
+                if let requestBody = maybeRequestBody {
+                    Logger.error("Could not create request to \(path) with body \(requestBody)")
+                } else {
+                    Logger.error("Could not create request to \(path) without body")
+                }
+                if let completionHandler = maybeCompletionHandler {
+                    self.operationDispatcher.dispatchOnMainThread {
+                        completionHandler(-1, nil, ErrorUtils.networkError(withUnderlyingError: ErrorUtils.unknownError()))
+                    }
+                }
+                return
+            }
+
+            let request = HTTPRequest(httpMethod: httpMethod,
+                                      path: path,
+                                      requestBody: maybeRequestBody,
+                                      authHeaders: authHeaders,
+                                      retried: retried,
+                                      urlRequest: urlRequest,
+                                      completionHandler: maybeCompletionHandler)
+
+            if performSerially && !retried {
+                if self.currentSerialRequest != nil {
+                    let logMessage =
+                        String(format: Strings.network.serial_request_queued, self.queuedRequests.count, httpMethod, path)
+                    Logger.debug(logMessage)
+                    self.queuedRequests.append(request)
+                    return
+                } else {
+                    Logger.debug(String(format: Strings.network.starting_request, httpMethod, path))
+                    self.currentSerialRequest = request
+                }
+            }
+
+            let logMessage = String(format: Strings.network.api_request_started,
```

Comment:
I think this can probably be on the same line, doesn't look like it is longer than 120 chars

Reply:
goes over by just a bit 😢